### DISCUSSION TITLE
Integrating with AWS Cognito & AppSync

### DISCUSSION THREAD

**Discussion by smlarkin** [urql-graphql/urql#2786] [Q&A]
Hello All,

I'm looking to use URQL as a client with a both AWS Cognito for auth & AWS AppSync for the graphql endpoint.

We are currently using AWS Amplify, which has been problematic.  Having some Apollo Client & recently URQL experience, I think it would be ideal to wire our AWS services up with URQL as the client.

Is there any starter apps, examples, etc that walk through this that someone might recommend ?

Cheers!

**Comment by mrflip** [comment]
Until I can find time to release a proper demonstration of a production React Native app for people to criticize and build on, here are the essential pieces:

https://gist.github.com/mrflip/1e9755a2d99dee197b2bd95e4ec67fe6

Glad to answer questions and to get feedback.

**Reply by smlarkin** [reply]
@mrflip Thank you.

This is most helpful for the Cognito part.  I am actually working on a proof of concept (hopefully to go into prod) using the new [Expo Router](https://expo.github.io/router/docs) in a Yarn workspace.

Do you have any time OR are you open to an exchange to look at how this is wired up ?
